File: go/Count_Primes.go
package main

import (
    "fmt"
)

func countPrimes(n int) int {
    var primes []bool = make([]bool, n)
    for i := 0; i < n; i ++ {
        primes[i] = true
    }
    for i := 2; i < n; i ++ {
        if primes[i] == true {
            for j := i * 2; j < n; j += i {
                primes[j] = false
            }
        }
    }
    res := 0
    for i := 2; i < n; i ++ {
        if primes[i] == true {
            res ++
        }
    }
    return res
}

func main() {
    fmt.Println(countPrimes(10))
}

File: js/Count_Primes.js
/**
 * @param {number} n
 * @return {number}
 */
var countPrimes = function(n) {
    if (n === 1) {
        return 0;
    }
    let res = 0;
    outer:
    for (let i = 2; i < n; i ++) {
        let l = Math.floor(Math.sqrt(i));
        for (let j = 2; j <= l; j ++) {
            if (i % j === 0) {
                continue outer;
            }
        }

        res ++;
    }
    return res;
};
// 我见青山多妩媚，料青山见我应如是
// 2,3,5,7
console.log(countPrimes(10));
